pymc_marketing/mmm/evaluation.py

```diff
@@ -0,0 +1,238 @@
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Evaluation and diagnostics for MMM models."""
+
+import arviz as az
+import numpy as np
+from sklearn.metrics import (
+    mean_absolute_error,
+    mean_absolute_percentage_error,
+    root_mean_squared_error,
+)
+
+from pymc_marketing.metrics import nmae, nrmse
+
+
+def calculate_metric_distributions(
+    y_true: np.ndarray,
+    y_pred: np.ndarray,
+    metrics_to_calculate: list[str] | None = None,
+) -> dict[str, np.ndarray]:
+    """Calculate distributions of evaluation metrics for posterior samples.
+
+    Parameters
+    ----------
+    y_true : np.ndarray
+        True values for the dataset. Shape: (date,)
+    y_pred : np.ndarray
+        Posterior predictive samples. Shape: (date, sample)
+    metrics_to_calculate : list of str or None, optional
+        List of metrics to calculate. Options include:
+            * `r_squared`: Bayesian R-squared.
+            * `rmse`: Root Mean Squared Error.
+            * `nrmse`: Normalized Root Mean Squared Error.
+            * `mae`: Mean Absolute Error.
+            * `nmae`: Normalized Mean Absolute Error.
+            * `mape`: Mean Absolute Percentage Error.
+        Defaults to all metrics if None.
+
+    Returns
+    -------
+    dict of str to np.ndarray
+        A dictionary containing calculated metric distributions.
+    """
+    metric_functions = {
+        "r_squared": lambda y_true, y_pred: az.r2_score(y_true, y_pred)["r2"],
+        "rmse": root_mean_squared_error,
+        "nrmse": nrmse,
+        "mae": mean_absolute_error,
+        "nmae": nmae,
+        "mape": mean_absolute_percentage_error,
+    }
+
+    if metrics_to_calculate is None:
+        metrics_to_calculate = list(metric_functions.keys())
+    else:
+        invalid_metrics = set(metrics_to_calculate) - set(metric_functions.keys())
+        if invalid_metrics:
+            raise ValueError(
+                f"Invalid metrics: {invalid_metrics}. "
+                f"Valid options are: {list(metric_functions.keys())}"
+            )
+
+    results = {}
+    for metric in metrics_to_calculate:
+        metric_values = np.array(
+            [
+                metric_functions[metric](
+                    y_true, y_pred[:, i]
+                )  # Calculate along date dimension
+                for i in range(y_pred.shape[1])
+            ]
+        )
+        results[metric] = metric_values
+
+    return results
+
+
+def summarize_metric_distributions(
+    metric_distributions: dict[str, np.ndarray] | None = None,
```

Comment:
remove the None case
```suggestion
    metric_distributions: dict[str, np.ndarray],